pep-9999.rst

```diff
@@ -0,0 +1,267 @@
+separation challenges from ``print()``.  However, this has two key problems.
+
+First, it changes the type of the exception, which is often a breaking change for
+downstream code.  We consider *always* raising a ``Wrapper`` exception unacceptably
+inelegant; but because custom exception types which might have any number of required
+arguments we can't always create an instance of the *same* type with our explanation.
+In cases where the exact exception type is known this can work, such as the standard
+library ``http.client`` code [7]_, but not for libraries which call user code.
+
+Second, exception chaining reports several lines of additional detail, which are
+distracting for experienced users and can be very confusing for beginners.
+For example, six of the eleven lines reported for this simple example relate to
+exception chaining, and are unnecessary with ``BaseException.__note__``:
+
+.. code-block:: python
+
+    class Explanation(Exception):
+        __module__ = "builtins"
+        def __str__(self) -> str:
+            return f"\n{self.args[0]}"
+
+    try:
+        raise AssertionError("Failed!")
+    except Exception as e:
+        raise Explanation("You can reproduce this error by ...") from e
+
+.. code-block:: python-traceback
+
+    $ python example.py
+    Traceback (most recent call last):
+    File "example.py", line 6, in <module>
+        raise AssertionError(why)
+    AssertionError: Failed!
+                                                                            # These lines are
+    The above exception was the direct cause of the following exception:    # confusing for new
+                                                                            # users, and they
+    Traceback (most recent call last):                                      # only exist due
+    File "example.py", line 8, in <module>                                  # to implementation
+        raise Explanation(msg) from e                                       # constraints :-(
+    Explanation:                                                            # Hence this PEP!
+    You can reproduce this error by ...
+
+
+Subclass Exception and add ``__note__`` downstream
+--------------------------------------------------
+Traceback printing is built into the C code, and reimplemented in pure Python in
+traceback.py. To get ``err.__note__`` printed from a downstream implementation
+would *also* require writing custom traceback-printing code; while this could
+be shared between projects and reuse some pieces of traceback.py we prefer to
+implement this once, upstream.
+
+Custom exception types could implement their ``__str__`` method to include our
+proposed ``__note__`` semantics, but this would be rarely and inconsistently
+applicable.
+
+
+Store notes in ``ExceptionGroup``s
+----------------------------------
+Initial discussions focussed on making a small change by thinking about how to
```

Comment:
```suggestion
Initial discussions focused on making a small change by thinking about how to
```

Reply:
Huh, Australian english rather than a typo for a change!  Thanks for all the copyedits 😍

Reply:
Oh sorry, wasn't aware that this is an accepted alternative spelling. Please use whatever variety of English you like :)